linkerd/distribute/src/lib.rs

```diff
@@ -0,0 +1,258 @@
+//! A load-agnostic traffic distribution stack module.
+
+#![deny(rust_2018_idioms, clippy::disallowed_methods, clippy::disallowed_types)]
+#![forbid(unsafe_code)]
+
+use indexmap::IndexMap;
+use linkerd_stack::{NewService, Param, Service};
+use rand::{
+    distributions::{Distribution as _, WeightedIndex},
+    rngs::SmallRng,
+    SeedableRng,
+};
+use std::{
+    collections::HashSet,
+    hash::Hash,
+    sync::Arc,
+    task::{Context, Poll},
+};
+
+#[derive(Clone, Debug)]
+pub struct NewDistribute<K, S> {
+    backends: Arc<IndexMap<K, S>>,
+}
+
+#[derive(Debug)]
+pub struct Distribute<K, S> {
+    backends: Arc<IndexMap<K, S>>,
+    ready: Option<S>,
```

Comment:
naming nit, take it or leave it: i might have called this `next`, to more clearly indicate that it's the service that will be called by the next `call`? alternatively, a comment here might make it clearer what this is for...